main.js

```diff
@@ -190,6 +234,27 @@ define(function (require, exports, module) {
 	addIcon('d', 'ion-contrast', '#960000');
 	addIcon('r', 'ion-ios-analytics', '#8495C0');
 
+	function extensionSearch(needle, haystack) {
```

Comment:
Remove this (line 237-257). Instead write a function that looks for a certain extension, by first checking whether it exists in `userIcons`, and if not, check it in `defaultIcons`